zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/StateControllerImpl.java

```diff
@@ -286,7 +286,7 @@ private void takeSnapshot(
 
   @Override
   public ActorFuture<PersistedSnapshot> takeSnapshot(
-      final int partition, final long lastProcessedPosition) {
+      final int partition, final long lastProcessedPosition, final ConcurrencyControl control) {
```

Comment:
:wrench: This is getting a bit confusing, especially because `takeTransientSnapshot` does not use the provided concurrencycontrol but the internal one. Can we make this simpler? Ideally we'd just use the internal cc.

Reply:
That was my first try as well, but unfortunately the internal CC is the `FileBaseSnapshotStore`. 
`TransientSnapshot.persist` uses `Actor.call` which cannot be used if you're already inside that CC :shrug: